src/networks.ts

```diff
@@ -1,19 +1,35 @@
+import fs from "fs";
+import path from "path";
+
 export type NetworkName = "testnet" | "testnet-sepolia" | "testnet-sepolia-staging" | "staging" | "localhost";
 
 export interface NetworkInfo {
   url: string;
   abi?: string;
 }
 
+let RPC_API_KEY: string | null = null;
+const keyPath = path.join(process.cwd(), ".earthfast", "rpc_api_key");
+
+if (fs.existsSync(keyPath)) {
+  RPC_API_KEY = fs.readFileSync(keyPath, "utf-8").trim();
+} else if (process.env.RPC_API_KEY) {
+  RPC_API_KEY = process.env.RPC_API_KEY;
+} else {
+  throw new Error(
+    "No RPC API key found. Please set it in .earthfast/rpc_api_key or as an environment variable RPC_API_KEY"
```

Comment:
i'm debating erroring out, i think maybe we should fallback to the ankr rpc so user can still continue the action. but lets print a message like "no alchemy API key provided falling back to ankr"

maybe you can make a rpcUrl variable above and default that to ankr and override with alchemy if api key is provided

Reply:
+1 to a fallback url